openpype/modules/deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -398,6 +405,153 @@ def inject_openpype_environment(deadlinePlugin):
+    """ Pull env vars from Ayon and push them to rendering process.
+
+        Used for correct paths, configuration from OpenPype etc.
+    """
+    job = deadlinePlugin.GetJob()
+
+    print(">>> Injecting Ayon environments ...")
+    try:
+        exe_list = get_ayon_executable()
+        exe = FileUtils.SearchFileList(exe_list)
+
+        if not exe:
+            raise RuntimeError((
+               "Ayon executable was not found in the semicolon "
+               "separated list \"{}\"."
+               "The path to the render executable can be configured"
+               " from the Plugin Configuration in the Deadline Monitor."
+            ).format(";".join(exe_list)))
+
+        print("--- Ayon executable: {}".format(exe))
+
+        ayon_bundle_name = job.GetJobEnvironmentKeyValue("AYON_BUNDLE_NAME")
+        if not ayon_bundle_name:
+            raise RuntimeError("Missing env var in job properties "
+                               "AYON_BUNDLE_NAME")
+
+        config = RepositoryUtils.GetPluginConfig("Ayon")
+        ayon_server_url = (
+                job.GetJobEnvironmentKeyValue("AYON_SERVER_URL") or
+                config.GetConfigEntryWithDefault("AyonServerUrl", "")
+        )
+        ayon_api_key = (
+                job.GetJobEnvironmentKeyValue("AYON_API_KEY") or
+                config.GetConfigEntryWithDefault("AyonApiKey", "")
+        )
+
+        if not all([ayon_server_url, ayon_api_key]):
+            raise RuntimeError((
+                "Missing required values for server url and api key. "
+                "Please fill in Ayon Deadline plugin or provide by "
+                "AYON_SERVER_URL and AYON_API_KEY"
+            ))
+
+        # tempfile.TemporaryFile cannot be used because of locking
+        temp_file_name = "{}_{}.json".format(
+            datetime.utcnow().strftime('%Y%m%d%H%M%S%f'),
+            str(uuid.uuid1())
+        )
+        export_url = os.path.join(tempfile.gettempdir(), temp_file_name)
+        print(">>> Temporary path: {}".format(export_url))
+
+        args = [
+            "--headless",
+            "extractenvironments",
+            export_url
+        ]
+
+        add_kwargs = {
+            "project": job.GetJobEnvironmentKeyValue("AVALON_PROJECT"),
+            "asset": job.GetJobEnvironmentKeyValue("AVALON_ASSET"),
+            "task": job.GetJobEnvironmentKeyValue("AVALON_TASK"),
+            "app": job.GetJobEnvironmentKeyValue("AVALON_APP_NAME"),
+            "envgroup": "farm",
+        }
+
+        if job.GetJobEnvironmentKeyValue('IS_TEST'):
+            args.append("--automatic-tests")
+
+        if all(add_kwargs.values()):
+            for key, value in add_kwargs.items():
+                args.extend(["--{}".format(key), value])
+        else:
+            raise RuntimeError((
+                "Missing required env vars: AVALON_PROJECT, AVALON_ASSET,"
+                " AVALON_TASK, AVALON_APP_NAME"
+            ))
+
+        os.environ["AVALON_TIMEOUT"] = "5000"
```

Comment:
This isn't probably needed in AYON as this sets timeout for mongodb connection if I am not mistaken.

Reply:
Ok, removed.